enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,377 @@
+
+## Alternatives
+
+### (Option A) Include the OpenShift Validator in upstream
+
+The proposal above was defined considering the new Validator be implemented in [operator-framework/api][oper-api] and used by SDK upstream implementation.
+
+**Pros**
+
+- Lower effort/cost since it would be similar to the others as it usage
+- Allow the Validator be used with OKD
+- The Validator can be used by any solution
+- Does not requires the usage/installation of a downstream SDK binary
+
+**NOTE** SDK users and Pipeline uses the upstream binary and shows that we have not a release process for SDK binary from downstream. Just for the images which are useful for Ansible/Helm operators.
+
+- Provide specific checks for OpenShift by default can help users which are from upstream and would like to publish their solutions on OperatorHub and OpenShift.
+- Users will still able to get the latest releases and the bug fixes faster and will not be affected for any delay that might can be faced to get the solutions from downstream.
+- Provide and promote the helper to the users since they can check that it is available by using the `--list-optional` flag:
+
+```sh
+$ operator-sdk bundle validate --list-optional
+NAME           LABELS                     DESCRIPTION
+operatorhub    suite=operatorframework    OperatorHub.io metadata validation
+               name=operatorhub          
+```
+
+**Cons**
+
+- Provide specific OpenShift solutions in upstream might hurt the community
+- OLM team would not like to add a code which is valid only for OpenShift unless we provide others vendor's validators as well in the public API. They are trying decouple the rules which are specific for OCP from OLM.
+- In the future it might result an effort required to move any OpenShift rules for downstream repos and implementations which are only valid for it. Currently, it would be the only case scenario so far, however, the precedence can bring others ones.
+
+### (Option B) Include the OpenShift Validator in downstream component for [operator-framework/api][oper-api] and use it in SDK upstream repository
+
+**Pros**
+
+- Mainly all pros of option A
+- Solves the OLM team concern about add OpenShift checks in [operator-framework/api][oper-api]
+  
+**Cons**
+
+- Effort required to keep a downstream repository for a downstream component and its releases [operator-framework/api][oper-api]
+- SDK would need to probably use only the downstream import to avoid misleading
+- It could cause delays for the SDK be able to apply the required changes of [operator-framework/api][oper-api] which can result ,for example, in an impact for Pipeline and users to get bug fixes or improvements.
+
+### (Option C) Design and implement Pluggable Validator mechanism for SDK
+
+Instead of we add the OpenshiftValidator to [operator-framework/api][oper-api] we would only provide it via the custom "downstream plugin" and make it available only for SDK binary.
```

Comment:
```suggestion
Instead of adding the OpenshiftValidator to [operator-framework/api][oper-api], we would provide it via the custom "downstream plugin" and make it available only for SDK binary.
```

Reply:
I don't think it would be "only for SDK binary"

Reply:
How another place would use the SDK Validator plugin? The solution suggested, SDK would define an interface for the CLI to be able to use the validators. Have you another option in mind?

Reply:
I think I saw here https://github.com/openshift/enhancements/pull/661#issuecomment-797588947. 👍 I think it is another option indeed.

Reply:
If the implementation is specified in an SDK plugin available only in the SDK downstream repo as suggested then, it cannot be used without SDK or imported by other projects.